.gitignore

```diff
@@ -1,6 +1,6 @@
 .DS_Store
 .project
 .vagrant/
-config/
-src/
```

Comment:
`src` is intended as the location to checkout the source of components when developing on an openBalena instance. For example, you could checkout the API, make changes and have them be reflected on the running openBalena instance. Removing the ignore rule means whatever is in that folder can potentially be committed into this repository.

I'm okay with moving haproxy and cert-provider into some folder, potentially `src` even, but we shouldn't forbid the usecase I'm describing. If you insist on reserving `src` for these two, maybe create a new folder (named `checkouts`? I don't know) and move the `src/README` file in there?

Reply:
After discussion, we decided we'll go with what we have and address the usecase I described separately in the future. Until then, we just have to be careful not to inadvertently commit a checkout into the repo.